### PR TITLE
Fixing flaky Cluster integration tests

### LINKED ISSUE #127: [BUG] testClusterGetSettingsWithDefault fails intermittently
**What is the bug?**

See https://github.com/opensearch-project/opensearch-java/runs/5645720812?check_suite_focus=true, tests failed on main against 1.3.0.

**What is the expected behavior?**
Tests pass.

### PR DESCRIPTION
**Pull request by VachaShah**
Signed-off-by: Vacha Shah <vachshah@amazon.com>

### Description
`testClusterGetSettings` and `testClusterGetSettingsWithDefault` fail intermittently. On debugging, found that the cluster settings contains `plugins={"index_state_management":{"metadata_migration":{"status":"1"},"template_migration":{"control":"-1"}}}` along with `cluster={"routing":{"allocation":{"enable":"NONE"}}}` during failures. Updating both the test assertions to fix the flaky test.
 
### Issues Resolved
Closes #127 
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dblock — COMMENTED** [review]
Also check that the value of `cluster` allocation setting is correct?

File: java-client/src/test/java/org/opensearch/client/opensearch/integTest/ClusterClientIT.java [hunk 1]
```diff
@@ -127,7 +127,7 @@ public void testClusterGetSettings() throws IOException {
 
         GetClusterSettingsResponse getSettingsResponse = openSearchClient.cluster()
                 .getSettings(new GetClusterSettingsRequest.Builder().build());
-        assertEquals(1, getSettingsResponse.persistent().size());
+        assertTrue(getSettingsResponse.persistent().containsKey("cluster"));
```
Comment (saratvemulapalli) [inline java-client/src/test/java/org/opensearch/client/opensearch/integTest/ClusterClientIT.java:130]: What does size return here? 
Does it return the number of keys present?  (I see the return type is `GetClusterSettingsResponse`)

Comment (VachaShah) [reply java-client/src/test/java/org/opensearch/client/opensearch/integTest/ClusterClientIT.java:130]: Yes it returns the size of the map

Comment (saratvemulapalli) [reply java-client/src/test/java/org/opensearch/client/opensearch/integTest/ClusterClientIT.java:130]: I agree checking on the cluster response.
Just curious, is the flakiness caused by the plugin sets the state sometimes and it doesnt at other times?

**Review by saratvemulapalli — APPROVED** [review]

**Review by VijayanB — APPROVED** [review]

File: java-client/src/test/java/org/opensearch/client/opensearch/integTest/ClusterClientIT.java (hunk 1 above)
Comment (VachaShah) [reply java-client/src/test/java/org/opensearch/client/opensearch/integTest/ClusterClientIT.java:130]: Yeah thats what I saw by running the test multiple times, the setting is present just some of the time.

**Pull request merged**